### ISSUE TITLE
Cannot apply 'aggregation_method = "event_timer"' alert condition

### ISSUE THREAD

**Issue by bearded-cat** [newrelic/terraform-provider-newrelic#1638]
- [x] Your New Relic `provider` [configuration](#terraform-configuration-files) (sensitive details redacted)
- [x] A list of [affected resources](#affected-resources) and/or data sources
- [x] The [configuration](#terraform-configuration-files) of the resources and/or data sources related to the bug report (i.e. from the list mentioned above)
- [x] Description of the [current behavior](#actual-behavior) (the bug)
- [x] Description of the [expected behavior](#expected-behavior)
- [x] Any related [log output](#debug-output)

### Terraform Version
```
Terraform v1.0.1
on darwin_amd64
+ provider [registry.terraform.io/newrelic/newrelic](http://registry.terraform.io/newrelic/newrelic) v2.36.2 (latest)
```

### Affected Resource(s)
- `newrelic_nrql_alert_condition`

### Actual Behavior

1. aggregation_method = "event_flow" alert condition  is set to NewRelic.

2. Apply(change) with the following terraform settings
```
  aggregation_method             = "event_timer"
  aggregation_timer                 = 60
(aggregation_delay is not included in this resource setting.)
```

3. following error occurs
Behaves as if a non-existent aggregation_delay is set. 

```
Error: INVALID_CONDITION: aggregationDelay: Aggregation delay cannot be set for aggregation method EVENT_TIMER: BAD_USER_INPUT
```

### Expected Behavior
No error.

### Steps to Reproduce

1. aggregation_method = "event_flow" alert condition  is set to NewRelic.

2. Apply(change) with the following terraform settings
```
  aggregation_method             = "event_timer"
  aggregation_timer                 = 60
(aggregation_delay is not included in this resource setting.)
```

**Comment by ronalcelaya** [comment]
Hi!
I'm having the same problem. 
`aggreagtion_delay` isn't included in the resource and terraform gives:
`Error: INVALID_CONDITION: aggregationDelay: Aggregation delay cannot be set for aggregation method EVENT_TIMER: BAD_USER_INPUT`

I think the issue exists since version 2.35.0. I have not problems using 2.34.1

**Comment by WUdDmfcf** [comment]
Thanks for the information!!
I'll try to work around it using 2.34.1 for now.
https://registry.terraform.io/providers/newrelic/newrelic/2.34.1/docs/resources/nrql_alert_condition

**Comment by bearded-cat** [comment]
If it's 2.34.1, this will work.

Case event_flow
```
  aggregation_method             = "event_flow"
  aggregation_delay              = 120
  aggregation_timer              = null
```

Case event_timer
```
  aggregation_method             = "event_timer"
  aggregation_delay              = null
  aggregation_timer              = 60
```

**Comment by emetcalf9** [comment]
I was able to reproduce this and put in a PR that will fix this issue.

This only happens if the previous aggregation_delay is 120 because we suppress the Diff when the default value (120) is used and no value is provided in Terraform. I have added a check on the aggregation_method so we only treat a missing value as matching the default if you are also using the related aggregation_method.

PR link here: https://github.com/newrelic/terraform-provider-newrelic/pull/1677
